pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -1035,4 +1037,52 @@ public ConsumingSegmentInfoReader.ConsumingSegmentsInfoMap getConsumingSegmentsI
           Response.Status.INTERNAL_SERVER_ERROR, e);
     }
   }
+
+  @POST
+  @Path("/segments/{tableNameWithType}/updateZKTimeInterval")
+  @Authenticate(AccessType.UPDATE)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Update the start and end time of the segments based on latest schema",
+      notes = "Update the start and end time of the segments based on latest schema")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"),
+      @ApiResponse(code = 404, message = "Table not found"),
+      @ApiResponse(code = 500, message = "Internal server error")
+  })
+  public SuccessResponse updateTimeIntervalZK(
+      @ApiParam(value = "Table name with type", required = true,
+          example = "myTable_REALTIME") @PathParam("tableNameWithType") String tableNameWithType) {
+      TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableNameWithType);
+      if (tableType == null) {
+        throw new ControllerApplicationException(LOGGER,
+            String.format("Table type not provided with table name %s", tableNameWithType),
+            Response.Status.INTERNAL_SERVER_ERROR);
+      }
+      return updateZKTimeIntervalInternal(tableNameWithType);
+  }
+
+  /**
+   * Internal method to update schema
+   * @param tableNameWithType  name of the table
+   * @return
+   */
+  private SuccessResponse updateZKTimeIntervalInternal(String tableNameWithType) {
+      TableConfig tableConfig = _pinotHelixResourceManager.getTableConfig(tableNameWithType);
+      Preconditions.checkState(tableConfig != null, "Failed to find table config for table: {}", tableNameWithType);
+
+      Schema tableSchema = _pinotHelixResourceManager.getTableSchema(tableNameWithType);
+      Preconditions.checkState(tableSchema != null, "Failed to find schema for table: {}", tableNameWithType);
+
+      String schemaName = tableSchema.getSchemaName();
+      try {
+        _pinotHelixResourceManager.updateSegmentsZKTimeInterval(tableConfig, tableSchema);
```

Comment:
Let's check the existence of time column, and get the `DateTimeFieldSpec` here. Throw proper exception (user error) when time column doesn't exist in table config, or `DateTimeFieldSpec` doesn't exist in schema. Only pass table name and dateTimeFieldSpec to the method